File: src/test/resources/oracle/commits/rm-action-test-c6e2f18c7ba5526bd698189c6b78f094fcd7b971/CustomerProfile.java
public class CustomerProfile {
    private String displayName;
    private Address address;
    private int loyaltyPoints;

    public CustomerProfile(String fullName, String street, String city, String postalCode, int loyaltyPoints) {
        this.displayName = fullName;
        this.address = new Address(street, city, postalCode);
        this.loyaltyPoints = loyaltyPoints;
    }

    public String mailingLabel() {
        return displayName + "\n" + address.format();
    }

    public boolean canReceivePromotion(boolean emailOptedIn, boolean bouncedEmail, int purchaseCount) {
        if (!emailOptedIn || bouncedEmail || purchaseCount <= 0) {
            return false;
        }
        return true;
    }

    public void addPurchase(int amountCents) {
        int earnedPoints = amountCents / 100;
        loyaltyPoints += earnedPoints;
    }

    public int getLoyaltyPoints() {
        return loyaltyPoints;
    }
}

File: src/test/resources/oracle/commits/rm-action-test-60d0d6f35eeed488560c247c3b4fa29dd3ac39d4/kotlin/CustomerProfile.kt
class CustomerProfile(
    fullName: String,
    street: String,
    city: String,
    postalCode: String,
    private var loyaltyPoints: Int
) {
    private val displayName: String = fullName
    private val address: Address = Address(street, city, postalCode)

    fun mailingLabel(): String {
        return "$displayName\n${address.format()}"
    }

    fun canReceivePromotion(emailOptedIn: Boolean, bouncedEmail: Boolean, purchaseCount: Int): Boolean {
        if (!emailOptedIn || bouncedEmail || purchaseCount <= 0) {
            return false
        }
        return true
    }

    fun addPurchase(amountCents: Int) {
        val earnedPoints = amountCents / 100
        loyaltyPoints += earnedPoints
    }

    fun getLoyaltyPoints(): Int {
        return loyaltyPoints
    }
}
